### ISSUE TITLE
Incorrect Unused varible warning when value from a map is used in a match pattern of nested fun

### ISSUE THREAD

**Issue by brigadier** [ignatov/intellij-erlang#938]
Example
```
test(Map, ListOfMaps) ->
	#{sort := Key} = Map,
	lists:sort(
		fun(#{Key := A}, #{Key := B}) ->
			A =< B
		end,
		ListOfMaps
	).
```

All theree instances of the `Key` vaiable are incorrectly highlighted yellow as 'Unused variable Key'.

**Comment by ignatov** [comment]
Could you please clarify that case. Do we use a value of the Key variable in the `fun(#{Key := A}, #{Key := B})`?

**Comment by brigadier** [comment]
Yeah we take value from a map and use it as the key. As far as I remember It didn't work in R17 but in one of the more recent version it is possible to do this. The function from my example is a correct one.
